### DISCUSSION TITLE
Only interested in same values from received serial string from PV warmwater heater ELWA

### DISCUSSION THREAD

**Discussion by kuhnino** [arendst/Tasmota#14415] [Q&A] [answered]
Hi there,
I have a solar powered warmwater heater system named ELWA ( https://www.my-pv.com/en/products/elwa)
When I connect the optional IR to USB-serial interface to my PC using termite I get following string back from ELWA

dr	V1.31	131	12	1	1	1	150	90	160	635	600	30	89	236.0	236.95	0.7459	176.74	423	245654	0	-25	3.7	**457**	219	199	0	1	1201002106080018	635	239

This is okay and works fine.
Now I want to replace the PC with an ESP8266 running tasmota. Hardwarevise I do not see any problems, but from the above mentioned string I´m only interested in a couple of data at present p.e. the highlighted 457 WhatsperDay.

Is there a way inside Tasmota to "select only" certain values and show them in the WebUI ?

FYI, the string details are

Field | Comment
-- | --
dr | Constant
vers | FW Version number
dayctr | Operating day counter
status | Operation mode, see Table 1
Rel_dcwp | DC Breaker status, 0: open, 1: closed
Rel_dc | DC Relay status, 0:open, 1: closed
Rel_ac | AC Relay status, 0: open, 1: closed
Tempist | Water temperature in 1/10°C
Tempmin | Current Day Temp min in 1/10°C
Tempmax | Current Day Temp max in 1/10°C
tempsoll_dc | DC Temp Setting in 1/10°C
tempsoll_ac | AC Temp Setting in 1/10°C
tempchip | Internal Temperature of Electronics in °c
iso_rel | DC Isolation value
UMpp | Internal Mpp value
UIst | DC Voltage in 0.00 V
Iist | DC Current in 0.0000 A
Pist | DC Power in 0.00 W
**Whperday | Current Day DC Energy in Wh**
Whtotal | Total DC Energy in Wh
Whac | Current Day AC Energy in Wh
Dp | Internal control value
Pwm | Internal control value
Scantimer | Internal control value
Minute | Internal clock
dc_ontime | Minutes since wakeup
ac_heatmode | AC switch on time, 0: Temp controlled
position | 0: lower unit, 1: upper unit
serialno | Device Serial number
pot_pos | Internal control value
checksum | 8 bit Checksum over entire reply string excluding checksum, including Tabs

**Comment by sfromis** [comment, marked as answer]
I'd recommend ESP32 instead, where you with Tasmota can get the Berry programming language, much more suitable for doing such parsing/display, you can basically create it as a sensor driver. Or with ESP8266, you could still create a driver in C++.

**Comment by kuhnino** [comment]
OK, good point, will give it a try. 

Any suggested ESP32 from your side ?

I´m a beginner, so better to buy a maybe higher priced device rather than a cheap one with some "failures or limitations", Quite sure you get the point ;-)

**Reply by sfromis** [reply]
Those NodeMCU-like ESP32 dev boards are not at all expensive, and nice to work with for tinkering. Unless you want features like integrated displays, RJ45, camera or battery management, not much reason to go above that. Maybe look for PSRAM if you envision bigger projects, but you can get quite some way without PSRAM and with the standard 4MB of flash.

Maybe be careful with the very cheap ESP32-CAM boards out there, people have had too much trouble with them.

**Comment by MiR893467** [comment]
Hello kuhnino,

did you have success in reading the data of you ELWA ?

I do have the same problem.
I have a DC ELWA, the USB adapter and try to read the data, NOT using the dumb windows tool.

I also tried to attach a simple serial terminal, but I do not get any data from the USB Adapter.

Regards

**Reply by MaRuDa55** [reply]
Hi,
i have written a Windows-App to get the data without the unusable original Windows tool from mypv. 
If you are interested you can find/download it here:
https://github.com/MaRuDa55/ELWA
Only in german.

**Reply by Noschvie** [reply]
@MaRuDa55 you download just provide an EXE file, that's risky...

**Comment by kuhnino** [comment]
Yes,
I did
Used nodered to read the usb serial data, convert them and forward them on Mqtt bus

If u let me know how to share, I can do that

Kuhnino

26.03.2023 15:39:18 MiR893467 ***@***.***>:

**Reply by Noschvie** [reply]
Why not using Tasmota for that?
